src/components/AutoComplete/AsyncAutoComplete.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { AsyncTypeahead } from 'react-bootstrap-typeahead';
+import PropTypes from 'prop-types';
+
+class AsyncAutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: []
+    };
+
+    this.onSearchHandler = this.onSearchHandler.bind(this);
+  }
+
+  onSearchHandler = () => {
+    const options = this.props.onSearch();
+    this.setState({ options });
+  };
+
+  render() {
+    return (
+      <AsyncTypeahead
+        {...this.props}
+        onSearch={this.onSearchHandler}
+        options={this.state.options}
+      />
+    );
+  }
+}
+
+AsyncAutoComplete.propTypes = {
+  onSearch: PropTypes.func.isRequired
```

Comment:
it looks like isLoading is also required. https://github.com/ericgio/react-bootstrap-typeahead/blob/master/docs/Props.md#asynctypeahead